providers/kind/kind.sh

```diff
@@ -0,0 +1,188 @@
+# This file is part of the Kepler project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2022 The Kepler Contributors
+#
+
+set -eu -o pipefail
+
+declare -r KIND_DIR="${KIND_DIR:-"$PROJECT_ROOT/tmp/kind"}"
+declare -r KIND="$BIN_DIR/kind"
+declare -r KIND_VERSION=${KIND_VERSION:-0.17.0}
+declare -r KIND_CONFIG_YAML="$KIND_DIR/kind.yml"
+
+declare -r KIND_REGISTRY_NAME=${KIND_REGISTRY_NAME:-kind-registry}
+declare -r KIND_REGISTRY_YAML="$KIND_DIR/local-registry.yml"
+
+declare -r KIND_DEFAULT_NETWORK="kind"
+declare -r KIND_CONFIG_PATH="$PROJECT_ROOT/providers/kind"
+declare -r KIND_MANIFESTS_DIR="$KIND_CONFIG_PATH/manifests"
+declare -r KIND_CLUSTER_NAME=${KIND_CLUSTER_NAME:-kind}
+declare -r KIND_IMAGE_REPO=${KIND_IMAGE_REPO:-localhost:5001}
+declare -r KIND_KUBECONFIG="$KIND_DIR/kubeconfig"
```

Comment:
Can we make this a common variable instead of cluster-type specific?

Reply:
I think this is a bit too premature to optimise. Currently you can have a `.env` that allows different values for each provider and I do see value in keeping provider specific variables on the other hand, you suggestion does make sense. How about we delay the decision until we have a concrete decision